Tests/SonarScanner.MSBuild.PreProcessor.Test/PreprocessorObjectFactoryTests.cs

```diff
@@ -39,21 +43,57 @@ public class PreprocessorObjectFactoryTests
             logger = new TestLogger();
 
         [TestMethod]
-        public void Factory_ThrowsOnInvalidInput()
+        public void CreateSonarWebService_ThrowsOnInvalidInput()
         {
             ((Func<PreprocessorObjectFactory>)(() => new PreprocessorObjectFactory(null))).Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("logger");
 
             var sut = new PreprocessorObjectFactory(logger);
-            sut.Invoking(x => x.CreateSonarWebService(null)).Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("args");
+            sut.Invoking(x => x.CreateSonarWebService(null).Result).Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("args");
+        }
+
+        [DataTestMethod]
+        [DataRow("https://sonarsource.com/", "https://sonarsource.com/api/server/version")]
+        [DataRow("https://sonarsource.com", "https://sonarsource.com/api/server/version")]
+        [DataRow("https://sonarsource.com/sonarlint", "https://sonarsource.com/sonarlint/api/server/version")]
+        [DataRow("https://sonarsource.com/sonarlint/", "https://sonarsource.com/sonarlint/api/server/version")]
```

Comment:
It is now the factory's responsibility to ensure the correct ending of the host URL. We check this here and simplify a lot the existing tests from `SonarWebService`.